File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java
import java.util.Arrays;
class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return birdsPerDay.clone();
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        ++birdsPerDay[birdsPerDay.length - 1];
    }

    public boolean hasDayWithoutBirds() {
        return Arrays.stream(birdsPerDay).anyMatch(v -> v == 0);
    }

    public int getCountForFirstDays(int numberOfDays) {
        return Arrays.stream(birdsPerDay).limit(numberOfDays).reduce(0, Integer::sum);
    }

    public int getBusyDays() {
        return (int)Arrays.stream(birdsPerDay).filter(v -> v >= 5).count();
    }
}

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay) => this.birdsPerDay = birdsPerDay;

    public static int[] LastWeek() => new int[] {0, 2, 5, 3, 7, 8, 4};

    public int Today() => birdsPerDay[birdsPerDay.Length - 1];

    public void IncrementTodaysCount() => ++birdsPerDay[birdsPerDay.Length - 1];

    public bool HasDayWithoutBirds() => !Array.TrueForAll(birdsPerDay, bird => bird != 0);

    public int CountForFirstDays(int numberOfDays)
    {
        var sum = 0;
        for (var i = 0; i < numberOfDays && i < birdsPerDay.Length; ++i)
        {
            sum += birdsPerDay[i];
        }
        return sum;
    }

    public int BusyDays()
    {
        var count = 0;
        foreach (var birds in birdsPerDay)
        {
            if (birds >= 5)
            {
                ++count;
            }
        }
        return count;
    }
}
